PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -160,6 +160,19 @@ PATMuonProducer::PATMuonProducer(const edm::ParameterSet & iConfig) :
     mvaEstimator_.initialize(fip.fullPath(),mvaDrMax_);
   }
 
+  // MC info
+  simClassification_        = consumes<edm::ValueMap<int> >(edm::InputTag("muonSimClassifier",""));
```

Comment:
why not turn these into a single data format ?

Reply:
Are you suggesting to put it all in a single container instead of a bunch of ints and floats? It can be done.

Reply:
consumes and use of these muonSim products better be made conditional on a config parameter.

Reply:
also, shouldn't at least the module instance be configurable instead of the hardcoded muonSimClassifier?